noop.go

```diff
@@ -68,7 +70,25 @@ func (n NoopTracer) StartSpanWithOptions(opts StartSpanOptions) Span {
 	return defaultNoopSpan
 }
 
+// Extractor belongs to the Tracer interface.
+func (n NoopTracer) Extractor(format interface{}) Extractor {
+	return defaultNoopInjectorExtractor
+}
+
+// Injector belongs to the Tracer interface.
+func (n NoopTracer) Injector(format interface{}) Injector {
+	return defaultNoopInjectorExtractor
+}
+
 // JoinTrace belongs to the Tracer interface.
 func (n NoopTracer) JoinTrace(operationName string, parent interface{}) Span {
 	return defaultNoopSpan
 }
+
+func (n noopInjectorExtractor) InjectSpan(span Span, carrier interface{}) error {
+	return nil
+}
+
+func (n noopInjectorExtractor) JoinTrace(operationName string, carrier interface{}) (Span, error) {
```

Comment:
Ah, I remember now. So the original Go API I wrote had this semantics:

```
span, err := tracer.joinTrace()
if err != nil or span == nil {
  span = tracer.startTrace()
}
```

Some people complained about this being too verbose and asking "why can't joinTrace do it in one line". I think you now discovered why - if we do it all in Join, we will have to swallow the error that may potentially occur during tracer ctx deserialization (malformed header or what not). The above code was explicit and leaving error handling to the caller, which is the Go way. So I think we need to decide on the exact semantics of the extractor's Join method. My inclination is to have it deal with missing header (span==nil above), but expose decoding error, so that people may alert or it, or emit metric.